metrics: Add workqueue metrics (#27042)
cilium-team/cilium: Documentation/observability/metrics.rst

``k8s_client_api_calls_total``                ``host``, ``method``, ``return_code``         Enabled    Number of API calls made to kube-apiserver labeled by host, method and return code
============================================= ============================================= ========== ===========================================================

Kubernetes workqueue
~~~~~~~~~~~~~~~~~~~~

================================================ ============================================= ========== ===========================================================
Name                                             Labels                                        Default    Description
================================================ ============================================= ========== ===========================================================
``workqueue_depth``                              ``name``                                      Enabled    Current depth of workqueue
``workqueue_adds_total``                         ``name``                                      Enabled    Total number of adds handled by workqueue

> ``workqueue_queue_duration_seconds``             ``name``                                      Enabled    How long in seconds an item stays in workqueue before being requested

Reviewer: Clarity nits
```suggestion
``workqueue_queue_duration_seconds``             ``name``                                      Enabled    Duration in seconds an item stays in workqueue prior to request
```

> ``workqueue_work_duration_seconds``              ``name``                                      Enabled    How long in seconds processing an item from workqueue takes

Reviewer: ```suggestion
``workqueue_work_duration_seconds``              ``name``                                      Enabled    Duration in seconds to process an item from workqueue
```

> ``workqueue_unfinished_work_seconds``            ``name``                                      Enabled    How many seconds of work has been done that is in progress and hasn't been observed by work_duration. Large values indicate stuck threads. One can deduce the number of stuck threads by observing the rate at which this increases

Reviewer: ```suggestion
``workqueue_unfinished_work_seconds``            ``name``                                      Enabled    Duration in seconds of work in progress that hasn't been observed by work_duration. Large values indicate stuck threads. You can deduce the number of stuck threads by observing the rate at which this value increases.
```

> ``workqueue_longest_running_processor_seconds``  ``name``                                      Enabled    How many seconds has the longest running processor for workqueue been running

Reviewer: ```suggestion
``workqueue_longest_running_processor_seconds``  ``name``                                      Enabled    Duration in seconds of the longest running processor for workqueue
```
